parachain/pallets/erc20-app/src/lib.rs

```diff
@@ -186,19 +186,23 @@ pub mod pallet {
 			Self::deposit_event(Event::Minted(token, sender, recipient.clone(), amount));
 
 			if let Some(destination) = destination {
-				let result = T::XcmReserveTransfer::reserve_transfer(
-					asset_id,
-					&recipient,
-					amount,
-					destination,
-				);
-				if let Err(err) = result {
-					log::error!(
-						"Failed to execute xcm transfer to parachain {} - {:?}.",
-						destination.para_id,
-						err
+				with_transaction(|| {
+					let result = T::XcmReserveTransfer::reserve_transfer(
+						asset_id,
+						&recipient,
+						amount,
+						destination,
 					);
-				}
+					if let Err(err) = result {
+						log::error!(
+							"Failed to execute xcm transfer to parachain {} - {:?}.",
+							destination.para_id,
+							err
+						);
+						return TransactionOutcome::Rollback(());
+					}
+					TransactionOutcome::Commit(())
+				});
```

Comment:
Since we can't return an error if the xcm transfer failed, we could emit an event instead, something like `XcmTransferFailed { error: XcmError }`.

Reply:
I have created a ticket for tracking XCM transfer state with events. SNO-227